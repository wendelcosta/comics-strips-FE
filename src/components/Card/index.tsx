import React from 'react'
import { makeStyles, createStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'

const useStyles = makeStyles((theme) =>
  createStyles({
    papers: {
      width: 250,
      height: 250,
      margin: '0 auto',
      '&:hover': {
        cursor: 'pointer ',
        boxShadow: `2px 5px 5px ${theme.custom.blue}`,
      },
    },
  })
)

interface CardProps {
  img: string
  onClickCard?(): void
}

const Card: React.FC<CardProps> = ({ onClickCard, img }) => {
  const classes = useStyles()
  return (
    <Paper className={classes.papers} onClick={onClickCard}>
      <img
        data-testid="card"
        src={img}
        alt="cards"
        className={classes.papers}
      />
    </Paper>
  )
}

export default Card
