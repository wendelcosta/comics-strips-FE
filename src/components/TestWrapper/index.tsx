import React from 'react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from '../../styles/theme'

const TestWrapper: React.FC = ({ children }: any) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export default TestWrapper
