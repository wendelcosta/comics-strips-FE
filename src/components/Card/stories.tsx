import React from 'react'

import Card from '.'
import TestWrapper from '../TestWrapper'

export default {
  title: 'Card',
  component: Card,
  argTypes: {
    img: { control: 'text' },
  },
}

export const Default = (args: Record<string, string>): any => (
  <TestWrapper>
    <Card img={args.img} {...args} />
  </TestWrapper>
)
Default.args = {
  img: 'https://imgs.xkcd.com/comics/stellar_evolution.png',
}
