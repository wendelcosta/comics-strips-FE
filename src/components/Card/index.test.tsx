import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'

import Card from '.'
import TestWrapper from '../TestWrapper'

describe('Card Test', () => {
  test('Loads Card', () => {
    const myMock = jest.fn()
    const { container } = render(
      <TestWrapper>
        <Card img="test" onClickCard={myMock} />
      </TestWrapper>
    )
    const card = screen.getByTestId('card')
    fireEvent.click(card)
    expect(myMock).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })
})
