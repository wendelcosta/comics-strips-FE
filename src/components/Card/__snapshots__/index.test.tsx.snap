// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test Loads Card 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-papers-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <img
      alt="cards"
      class="makeStyles-papers-1"
      data-testid="card"
      src="test"
    />
  </div>
</div>
`;
